src/apify_client/clients/resource_clients/run.py

```diff
@@ -226,6 +228,38 @@ def log(self: RunClient) -> LogClient:
             **self._sub_resource_init_options(resource_path='log'),
         )
 
+    def charge(
+        self: RunClient,
+        event_name: str,
+        count: int | None = None,
+        idempotency_key: str | None = None,
+    ) -> dict:
+        """Charge for an event of a Pay-Per-Event Actor run.
+
+        https://docs.apify.com/api/v2#/reference/actor-runs/charge-run/charge-run
+
+        Returns:
+            dict: Status and message of the charge event.
+        """
+        if not event_name:
+            raise ValueError('event_name is required for charging an event')
+
+        response = self.http_client.call(
+            url=self._url('charge'),
+            method='POST',
+            headers={
+                'idempotency-key': idempotency_key or f'{self.resource_id}-{event_name}-{int(time.time() * 1000)}',
```

Comment:
What if someone tries to charge for two occurences of one event twice in the same millisecond? Then the idempotency key will be the same and one charge will get ignored. Some unique key is needed here, I think a random string would be enough.

There's a `crypto_random_object_id` implementation in [Crawlee for Python](https://github.com/apify/crawlee-python/blob/master/src/crawlee/_utils/crypto.py#L21-L24), perhaps we could move it to [`apify-shared-python`](https://github.com/apify/apify-shared-python/blob/master/src/apify_shared/utils.py) and use it here.

Reply:
I took the idempotency key generation from https://github.com/apify-store/contact-info/blob/master/code/src/charging_manager.ts but it could be specific to their actors 🤔 I can move `crypto_random_object_id` 👍